### PR TITLE
[UPGRADE] Segment Analytics 2.13.0

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

Upgrade `@segment/analytics-react-native` from 2.9.0 to 2.13.0 and update patch.

- [Release notes `@segment/analytics-react-native` v2.9.0](https://github.com/segmentio/analytics-react-native/releases/tag/%40segment%2Fanalytics-react-native-v2.10.0)
- [Release notes `@segment/analytics-react-native` v2.10.1](https://github.com/segmentio/analytics-react-native/releases/tag/%40segment%2Fanalytics-react-native-v2.10.1)
- [Release notes `@segment/analytics-react-native` v2.11.0](https://github.com/segmentio/analytics-react-native/releases/tag/%40segment%2Fanalytics-react-native-v2.11.0)
- [Release notes `@segment/analytics-react-native` v2.12.0](https://github.com/segmentio/analytics-react-native/releases/tag/%40segment%2Fanalytics-react-native-v2.12.0)
- [Release notes `@segment/analytics-react-native` v2.13.0](https://github.com/segmentio/analytics-react-native/releases/tag/%40segment%2Fanalytics-react-native-v2.13.0)

**Screenshots/Recordings**

~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/372

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: patches/@segment+analytics-react-native+2.13.0.patch [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ 
+     await this.process(event);
+@@ -517,8 +519,14 @@ export class SegmentClient {
+       newUserId,
+     });
+ 
+-    await this.process(event);
+-    this.logger.info('ALIAS event saved', event);
++    const newTimestamp = new Date();
++    console.log(newTimestamp);
```
Comment (Cal-L) [inline patches/@segment+analytics-react-native+2.13.0.patch:37]: Remove console logs

File: patches/@segment+analytics-react-native+2.13.0.patch [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
++  anonymousId,
+ });
+ 
+ export const createScreenEvent = ({
+@@ -77,11 +83,11 @@ export const createAliasEvent = ({
+   previousId: userId || anonymousId,
+ });
+ 
+-export const applyRawEventData = (event: SegmentEvent): SegmentEvent => {
++export const applyRawEventData = (event: SegmentEvent, timestamp?: string): SegmentEvent => {
```
Comment (Cal-L) [inline patches/@segment+analytics-react-native+2.13.0.patch:90]:
```suggestion
+export const applyRawEventData = (event: SegmentEvent, timestamp: string = new Date().toISOString()): SegmentEvent => {
```

File: patches/@segment+analytics-react-native+2.13.0.patch [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+   return {
+     ...event,
+     messageId: getUUID(),
+-    timestamp: new Date().toISOString(),
++    timestamp: timestamp || new Date().toISOString(),
```
Comment (Cal-L) [inline patches/@segment+analytics-react-native+2.13.0.patch:95]:
```suggestion
+    timestamp,
```

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: patches/@segment+analytics-react-native+2.13.0.patch (hunk 1 above)
Comment (gantunesr) [reply patches/@segment+analytics-react-native+2.13.0.patch]: [12011c5](https://github.com/MetaMask/metamask-mobile/pull/5768/commits/12011c596043381d97e832d083b71bb3a9f1e683)

File: patches/@segment+analytics-react-native+2.13.0.patch (hunk 2 above)
Comment (gantunesr) [reply patches/@segment+analytics-react-native+2.13.0.patch]: [12011c5](https://github.com/MetaMask/metamask-mobile/pull/5768/commits/12011c596043381d97e832d083b71bb3a9f1e683)

**Review by Cal-L — APPROVED** [review]
Lgtm!

**Pull request merged**